packages/core/src/amazonqFeatureDev/session/sessionState.ts

```diff
@@ -321,7 +328,9 @@ export class CodeGenState extends CodeGenBase implements SessionState {
         public deletedFiles: DeletedFileInfo[],
         public references: CodeReference[],
         tabID: string,
-        private currentIteration: number
+        private currentIteration: number,
+        public codeGenerationRemainingIterationCount?: number,
```

Comment:
Consider putting public fields before private fields

Reply:
We couldn't because "A required parameter cannot follow an optional parameter."